### PR TITLE
OOP school library: set up associations

### PR DESCRIPTION
**Pull request by aneallaryea100**
# OOP school library: set up associations

- Create a class Classroom with the following:
- @label instance variable, should be initialized in the constructor.
- Setter and getter for @label (remember about attr_accessor).
- Create the has-many/belongs-to relationship between Classroom and Student. The following should be implemented:
- Create the has-many side (a classroom has many students).
- Create the belongs-to side (a student belongs to a classroom).
- Make sure that when adding a student to a classroom it also sets the classroom for the student.
- Make sure that when setting the classroom for a student it also adds it to the classrooms' students.
- Create a class Book with the following:
- @title and @author instance variables, should be initialized in the constructor.
- Setters and getters for instance variables (remember about attr_accessor).
- Create a class Rental with the following:
- @date instance variable, should be initialized in the constructor.
- Setter and getter for @date (remember about attr_accessor).
- Create the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental. The following should be implemented:
- Create the has-many side of Book and Rental (a book has many rentals).
- Create the belongs-to side of Rental and Book (a rental belongs to a book).
- Create the has-many side of Person and Rental (a person has many rentals).
- Create the belongs-to side of Rental and Person (a rental belongs to a person).
- Modify the constructor of Rental so Book and Person are set in it.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -2,14 +2,15 @@
 
 class Person < Nameable
   attr_accessor :name, :age
-  attr_reader :id
+  attr_reader :id, :rentals
 
   def initialize(age, name = 'Unknown', parent_permission: true)
```
Comment (GzimAsani) [inline person.rb:7]: - [ ]  kindly fix this typo `parent_submission = true`

File: person.rb [hunk 2]
```diff
@@ -2,14 +2,15 @@
 
 class Person < Nameable
   attr_accessor :name, :age
-  attr_reader :id
+  attr_reader :id, :rentals
```
Comment (GzimAsani) [inline person.rb]: - [ ] :rentals should be getter and setter kindly add it to `attr_accessor`

File: book.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
+class Book
+  attr_accessor :title, :author
+  attr_reader :rentals
```
Comment (GzimAsani) [inline book.rb]: - [ ] :rentals should be both getter and setter, so please include on the attr_accessor

File: book.rb [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  def add_rental(rental)
+    @rental.push(rental)
```
Comment (GzimAsani) [inline book.rb]: - [ ] Fix the typo, it should either be @rental or @rentals

File: book.rb [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+  end
+end
```
Comment (GzimAsani) [inline book.rb]: - [ ] A method for adding rentals should be present and it needs to include the following parameters(person and date):
```ruby
  def add_rental(person, date)
    # *the book itself should be sent as a parameter to create the new rental achieving the 'has-many' association. 
    Rental.new(date, self, person)
  end
```

File: rental.rb [hunk 6]
```diff
@@ -0,0 +1,14 @@
+class Rental
+  attr_accessor :date
```
Comment (GzimAsani) [inline rental.rb]: - [ ] :book and :person should be getter and setter, kindly include them on attr_accessor

File: person.rb [hunk 7]
```diff
@@ -2,14 +2,15 @@
 
 class Person < Nameable
```
Comment (GzimAsani) [inline person.rb:3]: A method for adding rentals should be present.: [lesson](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship). In order for a person to have many rentals 
 ### **person has many rentals**

Example: 
```ruby
  def add_rental(book, date)
    # *The person itself should be sent as a parameter to create the new rental achieving the 'has-many' association
    Rental.new(date, book, self)
  end
```

**Review by GzimAsani — CHANGES_REQUESTED** [review]
### Hello @aneallaryea100    :wave:   
 Good job so far! There are some issues that you still need to work on to go to the next project but you are almost there!
 
 ## Highlights
 * Good job with your project, it is almost done:+1:  
 
 ## Required Changes 

- [ ] Check the comment below :arrow_double_down: 






 ## Optional suggestions
 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding! clapclapclap
 
 Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
 
 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Mihndim2020 — APPROVED** [review]
# STATUS: APPROVED!!!  🎉 🎉 🎉 :champagne: :champagne: :champagne:
Hi @aneallaryea100, 

### Good job !!!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages
:heavy_check_mark: Descriptive Readme file

Your project is complete! Great work on making the changes requested by the previous reviewer. There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 :champagne: :champagne: :champagne:

Congratulations-Memes-FEATURED

## Optional suggestions

_Check the comments under the review._

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

### N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear or you can get to me on slack  using this handle @mihndim.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**